install_with_docker.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+
+sudo_cmd=""
+
+# Check sudo permissions
+if (( $EUID != 0 )); then
+    echo "===> Running installer with non-sudo permissions."
+    echo "     In case of any failure or prompt, run the script with sudo privileges."
+    echo ""
+else
+    sudo_cmd="sudo"
+fi
+
+os=""
+case "$(uname -sr)" in
+   Darwin*)
+     os="darwin" 
+     package_manager="brew" ;;
+   Linux*)
+     os="linux" 
+     package_manager="apt-get" ;;
+   Ubuntu*|Pop!_OS)
+     os="linux"
+     package_manager="apt-get" ;;
+   Amazon\ Linux*)
+     os="amazon linux"
+     package_manager="yum" ;;
+   Debian*)
+     os="linux"
+     package_manager="apt-get" ;;
+   Linux\ Mint*)
+     os="linux"
+     package_manager="apt-get" ;;
+   Red\ Hat*)
+     os="linux"
+     package_manager="yum" ;;
+   CentOS*)
+     os="linux"
+     package_manager="yum" ;;
+   *)
+     os="Not Found: $os_name"
+     echo 'Not Supported OS'
+	 exit 1
+     ;;
+esac
+
+arch=$(uname -m)
+if [[ $arch == x86_64* ]]; then
+    arch="amd64"
+elif  [[ $arch == arm* ]]; then
+    arch="arm64"
+else
+    echo 'Not Supported Architecture'
+fi
+
+is_command_present() {
+    type "$1" >/dev/null 2>&1
+}
+
+request_sudo() {
+    if hash sudo 2>/dev/null; then
+        echo -e "\n Need sudo access to complete the installation."
+        if (( $EUID != 0 )); then
+            sudo_cmd="sudo"
+            echo -e "Please enter your sudo password, if prompted."
+            if ! $sudo_cmd -l | grep -e "NOPASSWD: ALL" > /dev/null && ! $sudo_cmd -v; then
+                echo "Need sudo privileges to proceed with the installation."
+                exit 1;
+            fi
+
+            echo -e "Got Sudo access.\n"
+        fi
+	fi
+}
+
+
+install_docker() {
```

Comment:
this func should be called `install_docker_for_non_mac` or something, since the brew case is not handled here, and its confusing to why brew is not handled in this function, then below there is a check that calls this function only for non brew cases